Balenalib2

dansku opened the pull request:
Renaming from resin to balena

dansku committed:
Update resin to balena

dansku committed:
Rename resin to balena

Change-type: minor
Signed-off-by: Daniel Andrade <daniel@balena.io>

lurch commented:
I guess this fixes #84 ?

imrehg commented on CHANGELOG.md:
```diff
@@ -4,6 +4,11 @@ All notable changes to this project will be documented in this file
 automatically by Versionist. DO NOT EDIT THIS FILE MANUALLY!
 This project adheres to [Semantic Versioning](http://semver.org/).
 
+# v.0.1.1
```
Why is this edit in here? It should not be edited manually, as it is created by versionbot.

dansku replied:
Agreed!

imrehg commented:
@dansku where did you change things to `balenalib`? I don't see anything like that in the changed files, and that was already done in #88 ?

dansku commented:
@imrehg yeah I noticed that it was changed and not merged after I updated the files (if I remember correctly). But I also updated the readme and the files.

lurch commented on README.md:
```diff
@@ -1,35 +1,35 @@
-- add the _resin remote_ to your local workspace using the useful shortcut in the dashboard UI ![remoteadd](https://raw.githubusercontent.com/resin-io-playground/boombeastic/master/docs/gitresinremote.png)
-- `git push resin master`
+- Sign up on [balena](https://dashboard.balena-cloud.com/signup)
+- go throught the [getting started guide](https://www.balena.io/docs/learn/getting-started/raspberry-pi/nodejs/) and create a new application
```
throught -> through

lurch commented on app/main.js:
```diff
@@ -66,29 +66,29 @@ if (process.env.NODE_ENV === 'development') {
       case 'lock':
-        lockFile.lock('/tmp/resin/resin-updates.lock', (error) => {
-          event.sender.send('resin-update-lock', error);
+        lockFile.lock('/tmp/balena/balena-updates.lock', (error) => {
```
According to https://www.balena.io/docs/learn/deploy/release-strategy/update-locking/ the lockfile path should be `/tmp/balena/updates.lock` ? :eyes:

dansku commented on Dockerfile.template:
```diff
@@ -62,13 +68,92 @@ RUN install_packages \
+WORKDIR /usr/src/app
+
+# Install build dependencies
+RUN install_packages \
```
Oh, vim went crazy here. Fixing...

dansku committed:
Small fixes

Change-type: minor
Signed-off-by: Daniel Andrade <daniel@balena.io>

dansku commented:
My bad, just saw that I was on a wrong local repo as I was editing these files a while ago. Pushing the fixes now.

dansku committed:
Merge branch 'master' into balenalib2

lurch commented on app/package.json:
```diff
@@ -27,9 +27,9 @@
   "license": "Apache-2.0",
   "bugs": {
-    "url": "https://github.com/resin-io/resin-electronjs/issues"
+    "url": "https://github.com/balena-io/balena-electronjs/issues"
```
This URL doesn't work yet - I guess somebody with the necessary privileges will need to rename this repo after this PR is merged?

dansku replied:
Yeah, we could ask @shaunmulligan to do it I believe

shaunmulligan replied:
@dansku yeah can update it once merged, if you give me a ping.

dansku replied:
@shaunmulligan will do!

lurch replied:
ping @shaunmulligan I guess it's time for this repo to be renamed? (as this PR has been merged)

imrehg reviewed (approved):
LGTM, let's get this over with, sorry for the delay, @dansku

lurch commented on app/main.js:
```diff
@@ -66,29 +66,29 @@ if (process.env.NODE_ENV === 'development') {
+if (electronConfig.ELECTRON_BALENA_UPDATE_LOCK) {
   const lockFile = require('lockfile');
-  electron.ipcMain.on('resin-update-lock', (event, command) => {
+  electron.ipcMain.on('update-lock', (event, command) => {
```
I've just noticed that the example code in `README.md` is using e.g. `ipcRenderer.once('balena-update-lock', (event, error) => {`
Does it need to be updated to instead use the `'update-lock'` name that is being used in `main.js` ?
